### PR TITLE
Photos picker: Add logic to persist session in cookie

### PR DESCRIPTION
**Pull request by bogiii**
Related to pdtkmj-36s-p2#comment-5914

## Proposed Changes

* Add functions to `persist` and `retrieve` session object stored in the cookie
* Extend `createSession` condition, checking if the session has expired

## Why are these changes being made?

* To support persisting sessions:
  * if a user clicks on the browser's refresh button (Calypso and Jetpack)
  * persist session between Calypso and Jetpack

## Testing Instructions

* Make sure you have the Photos Picker feature turned ON
* Go to `/media/{SITE_SLUG}`
* Select "Google Photos" source
* Make some selection in the picker tab
* Click on the browser refresh button
* Go to Google Photos
* Verify if the previous selection is available

https://github.com/user-attachments/assets/a0121231-8098-409e-aadb-74d55d0a7a25

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by fditrapani — APPROVED** [review]
Tested as described and works! 

I don't think this is related to this PR but if I add an image in the media gallery, then navigate to the editor and add an image block, the images don't seem to carry through when I pick Google Photos there. I have to connect again. The same goes for connecting in the Editor then navigating to the Media gallery,

File: client/my-sites/media-library/google-photos-picker-button.tsx [hunk 1]
```diff
@@ -22,11 +24,18 @@ const GooglePhotosPickerButton = () => {
 	};
 
 	useEffect( () => {
-		! session && createSession();
-	}, [ session, createSession ] );
+		const isSessionExpired =
+			session?.expireTime && moment( session.expireTime ).isBefore( new Date() );
+
+		if ( ! session || isSessionExpired ) {
```
Comment (ouikhuan) [inline client/my-sites/media-library/google-photos-picker-button.tsx:30]: Not sure if this represents all the cases, should we also consider that session might be deleted in some cases and the session is not valid anymore?

**Review by ouikhuan — COMMENTED** [review]
Overall it looks good but we have an edge case here, to replicate it:
- Try to establish the Google Photos session first and have the media showing on WordPress.com.
- Refresh the media page and pick the Google Photos again, make sure the photo you selected previously show up correctly.
- Open your [developer console ](https://developer.wordpress.com/docs/api/console/) and make a request to delete your session `DELETE wpcom/v2//meta/external-media/session/google_photos/SESSION_ID`
- Go back to WordPress.com and refresh the page again.
- It'll stuck at the skeleton phase.

**Comment by bogiii** [comment]
Hey @ouikhuan,

Thanks for the review.
I have fixed the edge case you mentioned and also simplified the solution. I changed the way we cache the session. Now, we only store the session id.

This is a more reliable solution.

When you find some time, please take a look. Thanks a bunch!!!

**Review by ouikhuan — APPROVED** [review]
LGTM! Great move to store session id only in the COOKIE!

**Pull request merged**